### ISSUE TITLE
Integration into the ChemicalToolBox

### ISSUE THREAD

**Issue by bgruening** [oddt/oddt#3]
Integration of ODDT into the ChemicalToolBox would be awesome! CTB comes with a variety of tools and would nicely complement ODDT. Let me know if you are interested, I will work on this as time permits.

Thanks!
Bjoern

**Comment by mwojcikowski** [comment]
Hi Bjoern,

Sure, this seams to be nice! Let me know what kind of input do you need. I have never used Galaxy and ChemicalToolBox, although major dependencies are already in place.

Maciek

**Comment by bgruening** [comment]
@mwojcikowski I think most of the dependencies should be there. We also have defined many datatypes + converters.
It's just a matter of time. But this would be really awesome. We also have some nice protein and compound visualisations included https://github.com/bgruening/galaxytools/tree/master/visualisations.

Finally I can abandon my poorly written autodock integration :)

**Comment by mwojcikowski** [comment]
Ok. I'd have to get more familiar with Galaxy. What functions would you expect to be implemented in galaxy?

**Comment by bgruening** [comment]
We could start with `oddt.scoring` and `oddt.docking` this is really weak or not existent in CTB. Otherwise what ever is the easiest function to get us started :)

**Comment by mwojcikowski** [comment]
I guess all the functions would be easy, when I get my way around of creating nodes for Galaxy. I will start from those two modules you mentioned. Is there some appealing how to/tutorial you can recommend?

**Comment by bgruening** [comment]
Very good documentation is here: http://planemo.readthedocs.org/en/latest/

Thanks a lot!
Bjoern

**Comment by mwojcikowski** [comment]
Finally I managed to construct a CLI for ODDT (#13). Now it should be fairly easy to construct a XML file for Galaxy for rescoring purposes.

**Comment by bgruening** [comment]
This is awesome! Do you want me to work on this or are you planning to do this?

**Comment by mwojcikowski** [comment]
If you could assemble the initial XML it would be great. I would have to dig my way through Galaxy documentation since I've never done it.

I also plan to have docker containers, should we run it from docker? I would much prefer to have it there, and manage the dependencies in Docker via conda than recompile OB/RDKit within Galaxy. What do you think?

I need a day or two to write some documentation (yeah, the most "fun" part) and I will merge CLI to master and release 0.2 with CLI support.

**Comment by bgruening** [comment]
Ok, I will work on this during the weekend. Galaxy can handle conda dependencies natively, so no need for docker. But I have a chemicaltoolbox Docker Image and I plan to integrate this tool into it if you think this is a good idea.

So if you have a conda repo with all deps, this is all what we need! :)

**Comment by mwojcikowski** [comment]
Sure thing: https://anaconda.org/mwojcikowski There you have both openbabel and oddt. Openbabel is currently built from Git, although 2.4 is coming and I aim to compile it there, or help with official repo if it comes out. RDKit has its own official channel: https://anaconda.org/rdkit.

I used your Docker image and it works fine, although OB takes for ever to compile (as usual). I would go with Conda.

**Comment by bgruening** [comment]
Yeah, need to switch the image to conda - this will lower build times. If you want then contribute to bioconda. I think I have a openbabel build there.

**Comment by mwojcikowski** [comment]
I can't work with buggy and old 2.3.2 myself :) Although ODDT should work just fine. I will let you know how it goes with bioconda, although now we push every tag from github to conda repo automatically with Travis CI. This will not be as convenient with bioconda for sure.

**Comment by bgruening** [comment]
Sure as you prefer Galaxy can handle this. Thanks this is exciting!

**Comment by mwojcikowski** [comment]
Let me know if you need my help. FYI CLI can write to any OB file format and CSV (both to file and stdout).

**Comment by bgruening** [comment]
Yeah, saw this this is really neat. I might change CSV to TSV but this is easy on the Galaxy side. I will keep you posted. Very nice work!

**Comment by mwojcikowski** [comment]
Here are some docs for CLI:
http://oddt.readthedocs.org/en/latest/#oddt-command-line-interface-cli
